controllers/synapse/synapse_configmap.go

```diff
@@ -2677,9 +2677,33 @@ redis:
 	}
 
 	// Set Synapse instance as the owner and controller
-	ctrl.SetControllerReference(s, cm, r.Scheme)
+	if err := ctrl.SetControllerReference(s, cm, r.Scheme); err != nil {
+		return &corev1.ConfigMap{}, err
+	}
+
+	return cm, nil
+}
+
+func (r *SynapseReconciler) configMapForSynapseCopy(
+	s *synapsev1alpha1.Synapse,
+	objectMeta metav1.ObjectMeta,
+) (client.Object, error) {
+	var copyConfigMap *corev1.ConfigMap
+
+	sourceConfigMapName := s.Spec.Homeserver.ConfigMap.Name
+	sourceConfigMapNamespace := r.getConfigMapNamespace(*s, s.Spec.Homeserver.ConfigMap.Namespace)
+
+	copyConfigMap, err := r.getConfigMapCopy(sourceConfigMapName, sourceConfigMapNamespace, objectMeta)
+	if err != nil {
+		return &corev1.ConfigMap{}, err
+	}
```

Comment:
On the attempt of understanding the code here I see two functions pretty similar in name `configMapForSynapseCopy` and `getConfigMapCopy`. Some comments here can help clarify the differences between the two and turn the code more explicit. Changing the names may also make sense.

Reply:
I added some comments, let me know if that makes it clearer